File: Algorithms/118 - Pascal's Triangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        ans = []
        
        if numRows == 0:
            return []
        
        # row 1
        row = []
        row.append(1)
        ans.append(row)
        
        if numRows <= 1:
            return ans
        
        # row 2
        row = []
        row.append(1)
        row.append(1)
        ans.append(row)
        
        if numRows <= 2:
            return ans
            
        i = 2
        while i < numRows:
            j = 1
            row = [1]
            while j < i:
                row.append(ans[i - 1][j - 1] + ans[i - 1][j])
                j += 1
            row.append(1)
            ans.append(row)
            i += 1
        return ans
        
        

File: cpp/118 - Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        // var
        vector<int> tmp;
        vector<vector<int>> result;
        
        // run
        for (int i = 0; i < numRows; i++) {
            // add
            for (int j = 0; j < i + 1; j++)
                tmp.push_back((j == 0 || j == i) ? 1 : result[i - 1][j - 1] + result[i - 1][j]);
            result.push_back(tmp);
            tmp.clear();
        }
        return result;
    }
};